dask/dataframe/tests/test_dataframe.py

```diff
@@ -1634,13 +1634,22 @@ def test_combine():
 
     first = lambda a, b: a
 
+    # You can add series with strings and nans but you can't add scalars 'a' + np.NaN
+    str_add = lambda a, b: a + b if a is not np.nan else a
```

Comment:
It seems like `pd.Series.combine` behaves differently than `pd.DataFrame.combine`. For instance `df1 + df2` works and `df1.b + df2.b` works, but `df1.b[0] + df2.b[0]` might throw an error if one of the values is nan. `pd.Series.combine` with `add` runs into this last case.

Reply:
To make this concrete:

```python
import numpy as np
import pandas as pd
add  = lambda a, b: a + b

a = pd.Series(["a", "b", np.nan], dtype="O")
b = pd.Series(["a", "b", "c"])

a.combine(b, add)
```

```python-traceback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-27-49588c6c8e7b> in <module>
      6 b = pd.Series(["a", "b", "c"], dtype="O")
      7 
----> 8 a.combine(b, add)

~/conda/envs/dask-upstream/lib/python3.8/site-packages/pandas/core/series.py in combine(self, other, func, fill_value)
   2991                 rv = other.get(idx, fill_value)
   2992                 with np.errstate(all="ignore"):
-> 2993                     new_values.append(func(lv, rv))
   2994         else:
   2995             # Assume that other is a scalar, so apply the function for

<ipython-input-27-49588c6c8e7b> in <lambda>(a, b)
      1 import numpy as np
      2 import pandas as pd
----> 3 add  = lambda a, b: a + b
      4 
      5 a = pd.Series(["a", "b", np.nan], dtype="O")

TypeError: unsupported operand type(s) for +: 'float' and 'str'
```

Reply:
It seems like this is expected behavior given the docstring but it is a little unintuitive considering that this works fine if the series are cast to dataframes. @jorisvandenbossche can you confirm that this is expected?

Reply:
It's indeed a bit surprising inconsistency between Series and DataFrame, but so the underlying reason is that for Series the function is applied element-wise (so basically doing `np.nan + "c"`) while for DataFrame.combine the function is applied column-wise (so basically `a + b`, where for `Series.__add__` pandas can handle missing values). 

Now, what I am wondering: how did this change? (this seems behaviour in pandas that already exists for a longer time)

Reply:
I believe the change came from nans being interpreted properly in pandas rather than as the string "nan"

Reply:
If you run this test on dask main you end up with items like `"nanc"`